File: 16-3sum-closest/16-3sum-closest.py
# 호두마루님 코드

class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        closest = float("inf")

        for i in range(len(nums)-2):
            l,r = i+1, len(nums) - 1
            
            while l < r and (target - closest):
                v = nums[i] + nums[l] + nums[r]
                closest = closest if (abs(target - closest) < abs(target - v)) else v
                

                if v > target: r-= 1
                else: l += 1
        return closest

File: 16-3sum-closest/16-3sum-closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        sort(nums.begin(), nums.end());
        
        int ans = 0, dif = 0x3f3f3f3f;
        for (int i = 0; i < nums.size(); ++i) {
            int j = i + 1, k = nums.size() - 1;
            while (j < k) {
                int now = nums[i] + nums[j] + nums[k];
                if (dif > abs(now - target)) {
                    dif = abs(now - target);
                    ans = now;
                }
                
                if (now > target) --k;
                else ++j;
            }
        }
        
        return ans;
    }
};